bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeConverter.java

```diff
@@ -121,6 +126,9 @@ public static Object convertValues(Type targetType, Object inputValue) {
                 return anyToFloat(inputValue, () ->
                         ErrorUtils.createNumericConversionError(inputValue, PredefinedTypes.TYPE_FLOAT));
             case TypeTags.STRING_TAG:
+                if (inputValue instanceof BString) {
+                    return inputValue;
```

Comment:
Shall we add a test to cover this case?

Reply:
We can remove this check as per my understanding. This line checks for a `BString` value to convert it into `BString`. This will never happen as I checked we do a `checkIsType()` on the source and target types before coming to the switch case. Therefore it will never hit this line of code